pyQuARC/main.py

```diff
@@ -138,6 +138,60 @@ def _cmr_query(self):
             query = f"{orig_query}&page_num={page_num}"
 
         return concept_ids
+    
+    def _get_latest_version(self, concept_id):
+        """
+        Fetches the latest revision version for a given concept_id from CMR
+    
+        Args:
+            concept_id (str): The concept ID to query
+        
+        Returns:
+            str: The latest revision number, or None if not found
+        """
+        try:
+            # Construct the CMR metadata URL for the concept
+            url = f"{self.cmr_host}/search/concepts/{concept_id}.umm_json"
+            headers = get_headers()
+            response = requests.get(url, headers=headers)
+        
+            if response.status_code == 200:
+                # Extract revision-id from response headers
+                revision_id = response.headers.get('CMR-Revision-Id')
+                return revision_id
+            else:
+                print(f"Warning: Could not fetch latest version for {concept_id}. Using default.")
+                return None
+        except Exception as e:
+            print(f"Error fetching latest version for {concept_id}: {str(e)}")
+            return None
+
+    def _get_collection_version(self, concept_id):
+        """
+        Fetch the MetadataSpecification.Version of a collection from CMR.
+        Args:
+            concept_id (str): The concept ID to query.
+
+        Returns:
+            str: The collection's MetadataSpecification.Version, or None if not found.
+        """
+        try:
+            url = f"{self.cmr_host}/search/concepts/{concept_id}.umm_json"
+            headers = get_headers()
+            response = requests.get(url, headers=headers)
+
+            if response.status_code == 200:
+                data = response.json()
+                # UMM collections have MetadataSpecification.Version
+                version = data.get("MetadataSpecification", {}).get("Version")
+                return version
+            else:
+                print(f"Warning: Could not fetch metadata for {concept_id}.")
+                return None
+        except Exception as e:
+            print(f"Error fetching collection version for {concept_id}: {str(e)}")
+            return None
+
```

Comment:
The two methods have a lot in common - can you DRY (dont repeat yourself) it? also not sure what's the difference between these two versions and it looks like you're just printing the collection_version, but not using it elsewhere, do you even need that?